docs/tools/platformio.mdx

```diff
@@ -0,0 +1,124 @@
+
+## Luos configuration for _platformio.ini_
+
+Some build flags definition may be used to configure Luos engine when it is added in your project. Those definitions will take care of the Luos engine library by using a Luos script called when building a project with Luos engine. Those definitions must be set in _platformio.ini_.
+For more information, see the script _source_filter_script.py_ in <a href="https://github.com/Luos-io/luos_engine" target="_blank" rel="external nofollow">Luos engine <IconExternalLink width="10" /></a>.
+
+### Definition for Luos engine Library
+
+| Definition  |                                            Description |                                                                           Value                                                                            |
+| :---------: | -----------------------------------------------------: | :--------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   LUOSHAL   | Defines the HAL family to use with Luos engine library | <a href="https://github.com/Luos-io/luos_engine/tree/main/engine/HAL" target="_blank" rel="external nofollow"> Luos HAL <IconExternalLink width="10" /></a> |
+| NOTELEMETRY |                                     Disables telemetry |                                                                       On by default                                                                        |
+
+Every Luos project needs to have a _node_config.h_ file allowing to configure Luos engine's hardware dependencies and configure the library. See the [basics concept page](/docs/luos-technology/basics/orga) for more information.
+
+Here is a _platformio.ini_ project example:
+
+```AsciiDoc
+[env:nucleo_f072rb]
+platform = ststm32
+board = nucleo_f072rb
+framework = stm32cube
+
+build_unflags = -Os
+build_flags =
+    -include node_config.h
+    -DLUOSHAL=STM32F0
+    -O1
+lib_deps =
+    luos_engine
+```
+
+This project in the examples is Luos engine dependent. PlatformIO will download the last version of the library Luos engine from its registry. The file _node_config.h_ allowing users to configure Luos parameters and the hardware definition to be added to the project.
+
+<div align="center">
+  <Image src="/img/telemetry.png" darkSrc="/img/telemetry.png" />
+</div>
+
+You must define the <a href="https://github.com/Luos-io/luos_engine/tree/main/engine/HAL" target="_blank" rel="external nofollow">HAL family <IconExternalLink width="10" /></a> corresponding to your MCU. See the [compatibility page](/docs/compatibility/mcu_demoboard) to find the HAL family you must use for your MCU. During the build, a message appearing in red means a problem has occurred. It happens if you forget a build flag in your the file _platformio.ini_.
```

Comment:
We talked about the HAL thing previously, we should keep it together and avoid dispatching information on the entire document.

Reply:
place above